source/fab/language/fortran.py

```diff
@@ -204,6 +283,19 @@ def analyse(self, filename: Path) -> None:
                     logger.debug('Found %s called "%s"', unit, name)
                     self._state.add_fortran_program_unit(name, filename)
                     scope.append((unit, name))
+                    continue
+
+            match: Match = self._use_pattern.match(line)
+            if match:
+                name: str = match.group(3).lower()
+                if name in self._intrinsic_modules:
+                    logger.debug('Ignoring intrinsic module "%s"', name)
+                else:
+                    if len(scope) == 0:
+                        message = '"use" statement found outside program unit'
```

Comment:
Not sure about this. The static type tools can obviously infer that it is a string since that's what we assign to it. However I have done it so we can see what it looks like.

Reply:
I take your point here actually - it does make sense to try and avoid clutter where it is very obvious what the type being set is

Reply:
I've taken it out again.